CMake/Util.cmake

```diff
@@ -17,26 +17,104 @@ endfunction ()
 
 
 
-# message wrapper for warnings.
-macro (carla_warning)
+# message() wrapper for warnings.
+function (carla_warning)
   message (WARNING ${ARGN})
-endmacro ()
+endfunction ()
 
 
 
-# message wrapper for errors.
+# message() wrapper for errors.
 function (carla_error)
   message (FATAL_ERROR ${ARGN})
 endfunction ()
 
 
 
+function (carla_get_option_docs OUT_VAR)
+  get_property (DOCS GLOBAL PROPERTY CARLA_OPTION_DOCS)
+  set (${OUT_VAR} ${DOCS})
+  return (PROPAGATE ${OUT_VAR})
+endfunction ()
+
 macro (carla_option NAME DESCRIPTION VALUE)
   option (${NAME} ${DESCRIPTION} ${VALUE})
   carla_message ("(option) ${NAME} : ${VALUE}")
+  get_property (DOCS GLOBAL PROPERTY CARLA_OPTION_DOCS)
+  string (
+    APPEND
+    DOCS
+    "- ${NAME}\n"
+    "\t- Description: ${DESCRIPTION}\n"
+    "\t- Default: ${VALUE}\n"
+  )
+  set_property (GLOBAL PROPERTY CARLA_OPTION_DOCS ${DOCS})
 endmacro ()
 
+
+
 macro (carla_string_option NAME DESCRIPTION VALUE)
   set (${NAME} "${VALUE}")
   carla_message ("(option) ${NAME} : \"${VALUE}\"")
+  get_property (DOCS GLOBAL PROPERTY CARLA_OPTION_DOCS)
+  string (
+    APPEND
+    DOCS
+    "- ${NAME}\n"
+    "\t- Description: ${DESCRIPTION}\n"
+    "\t- Default: \"${VALUE}\"\n"
+  )
+  set_property (GLOBAL PROPERTY CARLA_OPTION_DOCS ${DOCS})
 endmacro ()
+
+
+
+function (carla_get_target_docs OUT_VAR)
```

Comment:
It looks good.

Just I want to add that Cmake ExternalProjects are also considered targets but are not documented. So we have Easter egg targets like:
```
cmake --build Build --target fastdds
```

Or this rare one added by Eigen:
```
cmake --build Build --target blas
```

What do you think about it? Makes sense to add also ExternalProject to the target doc list?
Probably, ExternalProject targets would add a lot of noise as there is no way at cmake to make targets private or not accessible.